src/pcg/grid.cpp

```diff
@@ -0,0 +1,44 @@
+#include <glog/logging.h>
+#include "src/pcg/grid.h"
+
+Grid::Grid(size_t width, size_t height)
+        : width(width), height(height) {
+    const size_t size = width * height;
+    array = std::make_unique<GridElement[]>(size);
+    if (!array) {
+        LOG(ERROR) << "Could not allocate space for grid of size: " << width << " x " << height;
+    }
+
+    std::fill(array.get(), array.get() + size, GridElement::NONE);
+}
+
+void Grid::fill(const BoundingBox2i& box, GridElement element) {
```

Comment:
I'm totally ok with changing this method depending on how it is supposed to work when it comes to box boundaries exlclusion